### PR TITLE
Convert `markdown-checker.js` to TS

### PR DESCRIPTION
**Pull request by vdusart**
## Description

Convert the `markdown-checker.js` file to TS.

## Related Issue
Close: #7454

### CODE REVIEW

**Review by pettinarip — CHANGES_REQUESTED** [review]
Nice @vdusart. Thanks for the PR. It was the latest `.js` file! 😄 

Things that are missing here:
- update the script under `package.json`. You can use `ts-node -O '{ \"module\": \"commonjs\" }' src/scripts/,,,`
- change the script name to camelCase to be consistent with the rest of the scripts

File: src/scripts/markdownChecker.ts [hunk 1]
```diff
@@ -39,11 +39,18 @@ const CASE_SENSITVE_SPELLING_MISTAKES = ["Thereum", "Metamask", "Github"]
 // Regex for explicit lang path (e.g. /en/) && for glossary links (trailing slash breaks links e.g. /glossary/#pos/ doesn't work)
 // We should have case sensitive spelling mistakes && check they are not in links.
 
-const langsArray = fs.readdirSync(PATH_TO_INTL_MARKDOWN)
+interface Languages {
+  lang?: Array<string>
```
Comment (corwintines) [inline src/scripts/markdownChecker.ts:43]:
```suggestion
  lang?: Array<Lang>
```

I was testing this out, and I think you can use this if you import the Lang type

```
import { Lang } from "../utils/languages"
```
Can use this if its any help.

Comment (vdusart) [reply src/scripts/markdownChecker.ts:43]: Thanks @corwintines :rocket:

File: src/scripts/markdownChecker.ts [hunk 2]
```diff
@@ -39,11 +41,18 @@ const CASE_SENSITVE_SPELLING_MISTAKES = ["Thereum", "Metamask", "Github"]
 // Regex for explicit lang path (e.g. /en/) && for glossary links (trailing slash breaks links e.g. /glossary/#pos/ doesn't work)
 // We should have case sensitive spelling mistakes && check they are not in links.
 
-const langsArray = fs.readdirSync(PATH_TO_INTL_MARKDOWN)
+interface Languages {
+  lang?: Array<Lang>
+}
+
+const langsArray: Array<string> = fs.readdirSync(PATH_TO_INTL_MARKDOWN)
```
Comment (corwintines) [inline src/scripts/markdownChecker.ts:48]:
```suggestion
const langsArray: Array<Lang> = fs.readdirSync(PATH_TO_INTL_MARKDOWN)
```

File: src/scripts/markdownChecker.ts [hunk 3]
```diff
@@ -65,36 +74,45 @@ function getAllMarkdownPaths(dirPath, arrayOfMarkdownPaths = []) {
...
   return languages
 }
 
-function processFrontmatter(path, lang) {
-  const file = fs.readFileSync(path, "utf-8")
-  const frontmatter = matter(file).data
+interface MatterData {
+  title: string
+  description: string
+  lang: string
```
Comment (corwintines) [inline src/scripts/markdownChecker.ts:107]:
```suggestion
  lang: Lang
```

**Review by corwintines — CHANGES_REQUESTED** [review]
@vdusart noticed a couple other spots for `Lang`.

Other than this, LGTM!

**Review by pettinarip — APPROVED** [review]
Thanks @vdusart

**Review by corwintines — APPROVED** [review]
Thanks @vdusart!

**Pull request merged**